.github/workflows/cypress-workflow-bundle-snapshot-based.yml

```diff
@@ -8,83 +8,25 @@ on:
-      - name: Get OpenSearch-Dashboards
-        run: |
-          wget https://ci.opensearch.org/ci/dbc/distribution-build-opensearch-dashboards/${{ env.VERSION }}/latest/linux/x64/tar/dist/opensearch-dashboards/opensearch-dashboards-${{ env.VERSION }}-linux-x64.tar.gz
-          tar -xzf opensearch-dashboards-${{ env.VERSION }}-linux-x64.tar.gz
-      - name: Get node and yarn versions
-        id: versions
-        run: |
-          echo "::set-output name=node_version::$(node -p "(require('./opensearch-dashboards-${{ env.VERSION }}-linux-x64/package.json').engines.node).match(/[.0-9]+/)[0]")"
-      - name: Setup node
-        uses: actions/setup-node@v1
-        with:
-          node-version: ${{ steps.versions.outputs.node_version }}
-          registry-url: 'https://registry.npmjs.org'
-      - name: Run OpenSearch-Dashboards server
-        run: |
-          cd opensearch-dashboards-${{ env.VERSION }}
-          bin/opensearch-dashboards serve &
-          timeout 300 bash -c 'while [[ "$(curl -s -u admin:admin -k localhost:5601/api/status | jq -r '.status.overall.state')" != "green" ]]; do sleep 5; done'
-      - name: Get Cypress version
-        id: cypress_version
-        run: |
-          echo "::set-output name=cypress_version::$(cat ./monetery-test/package.json | jq '.devDependencies.cypress' | tr -d '"')"
-      - name: Cache Cypress
-        id: cache-cypress
-        uses: actions/cache@v1
+      - uses: dorny/paths-filter@v2
+        id: filter
         with:
-          path: ~/.cache/Cypress
-          key: cypress-cache-v2-${{ runner.os }}-${{ hashFiles('**/package.json') }}
-        env:
-          CYPRESS_INSTALL_BINARY: ${{ steps.cypress_version.outputs.cypress_version }}
-      - run: npx cypress cache list
-      - run: npx cypress cache path
-      - name: Cypress tests
-        uses: cypress-io/github-action@v2
-        with:
-          working-directory: monetery-test
-          command: yarn cypress:run-with-security --browser chromium --spec 'cypress/integration/core-opensearch-dashboards/opensearch-dashboards/*.js'
-          wait-on: 'http://localhost:5601'
-      # Screenshots are only captured on failure, will change this once we do visual regression tests
-      - uses: actions/upload-artifact@v1
-        if: failure()
-        with:
-          name: cypress-screenshots
-          path: monetery-test/cypress/screenshots
-      # Test run video was always captured, so this action uses "always()" condition
-      - uses: actions/upload-artifact@v1
-        if: always()
-        with:
-          name: cypress-videos
-          path: monetery-test/cypress/videos
+          filters: |
+            tests:
+              - 'cypress/**/core-opensearch-dashboards/**'
+
+  tests:
+    needs: changes
+    if: ${{ needs.changes.outputs.tests == 'true' }}
+    uses: ./.github/workflows/release-e2e-workflow-template.yml
```

Comment:
I moved the bundle snapshot workflow to use the template since they were identical. The only difference was that the bundle workflow user `monetery-test` instead of `cypress-test`. I assume that this change is safe.Additionally the template also allows us to skip tests if there are no changes to the spec files that it works on, speeding up the CI

Reply:
Thanks for the refactoring!